app/public/js/localSprites.js

```diff
@@ -209,7 +205,7 @@ function addHomingFireballSprite(xPosition, yPosition, target) {
         $.extend(rectangle(4 * 32, 0 * 32, 32, 32), {image: fireballBImage, hitBox}),
     ];
     var homingFireballSprite = new SimpleSprite({frames}, xPosition, yPosition, 0, 0, 1.5, 1.5);
-    homingFireballSprite.name = 'homingFireball';
+    homingFireballSprite.type = 'SPRITE_TYPE_FIREBALL_HOMING';
```

Comment:
Ah so here this should be: `SPRITE_TYPE_FIREBALL_HOMING` (I would actually suggest `SPRITE_TYPE_HOMING_FIREBALL`)
And then somewhere else, perhaps the top of this file, you'll want to have many lines like:
```
var SPRITE_TYPE_FIREBALL_TRIGGER = 'fireballTrigger';
var SPRITE_TYPE_HOMING_FIREBALL = 'homingFireball';
var SPRITE_TYPE_FORCE_TRIGGER = 'forceTrigger';
```